spec/migration/migrators/delivery_partner_spec.rb

```diff
@@ -1,65 +1,33 @@
 RSpec.describe Migrators::DeliveryPartner do
-  # TODO: would be nice to use the 'it_behaves_like "a migrator"' shared_example but this one is difficult to make fail
-  # so the failure handling specs fail
-
-  describe '.record_count' do
-    it 'returns the count of delivery partners' do
-      FactoryBot.create_list(:migration_delivery_partner, 2)
-      expect(described_class.record_count).to eq(2)
-    end
-  end
-
-  describe '.model' do
-    it 'returns :delivery_partner' do
-      expect(described_class.model).to eq(:delivery_partner)
+  it_behaves_like "a migrator", :delivery_partner, [] do
+    def create_migration_resource
+      FactoryBot.create(:migration_delivery_partner)
     end
-  end
 
-  describe '.delivery_partners' do
-    it 'returns all delivery_partners' do
-      delivery_partner = FactoryBot.create(:migration_delivery_partner)
-      expect(described_class.delivery_partners).to include(delivery_partner)
+    def create_resource(migration_resource)
     end
-  end
 
-  describe '.reset!' do
```

Comment:
I'm not very familiar with this code, are `.reset!` and `delivery_partners` tested somewhere else? 

I couldn't see them in the shared examples.